internal/pkg/store/errors.go

```diff
@@ -48,3 +48,25 @@ func (e *ErrNoSuchEnvironment) Error() string {
 	return fmt.Sprintf("couldn't find environment %s in the project %s",
 		e.EnvironmentName, e.ProjectName)
 }
+
+// ErrApplicationAlreadyExists means that an application is already created in a specific project.
+type ErrApplicationAlreadyExists struct {
+	ApplicationName string
+	ProjectName     string
+}
+
+func (e *ErrApplicationAlreadyExists) Error() string {
```

Comment:
What do you think of enforcing adding the `Is` method to our custom errors?
```go
func (e *ErrApplicationAlreadyExists) Is(target error) bool {
    t, ok := target.(*ErrApplicationAlreadyExists)
    if !ok {
        return false
    }
    return e.ApplicationName == t.ApplicationName && e.ProjectName == t.ProjectName
} 
```
This way we can leverage `errors.Is(err, &ErrApplicationAlreadyExists{ApplicationName: "myapp", ProjectName: "myproject"}`  
It also makes unit testing easier.

Reply:
Oky doky!

Reply:
I did update the code, but I kept the unit tests using `require.EqualError` since not all the errors returned are our modeled errors (some are just wrapped fmt.Errors).

Unfortunately, the below error assertion fails.
```
require.True(t, 
    errors.Is(fmt.Errorf("get application api in project chicken: broken"),
              fmt.Errorf("get application api in project chicken: broken")))
```